ksqldb-parser/src/main/java/io/confluent/ksql/parser/tree/WithinExpression.java

```diff
@@ -33,32 +34,50 @@
   private final TimeUnit beforeTimeUnit;
   private final TimeUnit afterTimeUnit;
   private final JoinWindows joinWindows;
+  private final Optional<WindowTimeClause> gracePeriod;
 
   public WithinExpression(final long size, final TimeUnit timeUnit) {
     this(size, size, timeUnit, timeUnit);
   }
 
+  public WithinExpression(final long size, final TimeUnit timeUnit,
+                          final WindowTimeClause gracePeriod) {
+    this(Optional.empty(), size, size, timeUnit, timeUnit, Optional.of(gracePeriod));
```

Comment:
nit: this is inconsistent with the constructor above. We should either update the constructor above so that all constructors call the one with all arguments, or we should update this constructor to parallel the one above by having this one call the one with `before` and `after` but without `location`.